Assets/Scripts/Manager/DataManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class DataManager : MonoBehaviour
+{
+    public static DataManager Instance;
+    
+    private const string DATA_PATH = "Data";
+    private const string PLAYER_JSON = "Player";
+    private const string ENEMY_JSON = "Enemy";
+
+    private Dictionary<int, Player> _players = new Dictionary<int, Player>();
+    private Dictionary<int, Enemy> _enemies = new Dictionary<int, Enemy>();
+    
+    [System.Serializable]
+    public class Character
+    {
+        public int id;
+        public string charName;
+        public int level;
+        public int hp;
+        public int speed;
+        public int attackStat;
+    }
+    
+    [System.Serializable]
+    public class Player : Character
+    {
+        public ElementType elem;
+    }
+    
+    [System.Serializable]
+    public class Enemy : Character
+    {
+        public int shield;
+        public ElementType[] elem;
+    }
+
+    private struct PlayerData
+    {
+        public Player[] player;
+    }
+
+    private struct EnemyData
+    {
+        public Enemy[] enemy;
+    }
+
+    private void InitPlayerData()
+    {
+        TextAsset playerJson = Resources.Load<TextAsset>(Path.Combine(DATA_PATH, PLAYER_JSON));
+        PlayerData playerList = JsonUtility.FromJson<PlayerData>(playerJson.text);
+
+        foreach (var data in playerList.player)
+        {
+            var player = data as Character;
+            _players.Add(data.id, data);
+        }
+    }
+
+    private void InitEnemyData()
+    {
+        TextAsset enemyJson = Resources.Load<TextAsset>(Path.Combine(DATA_PATH, ENEMY_JSON));
+        EnemyData enemyList = JsonUtility.FromJson<EnemyData>(enemyJson.text);
+
+        foreach (var data in enemyList.enemy)
+        {
+            var enemy = data as Character;
+            _enemies.Add(enemy.id, data);
+        }
+    }
+
+    private void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+        
+        InitPlayerData();
+        InitEnemyData();
+    }
```

Comment:
```suggestion
    private void Awake()
    {
        if (Instance == null)
        {
            Instance = this;
        }
        else
        {
            Destroy(gameObject);
        }
        
        DontDestroyOnLoad(gameObject);
        InitPlayerData();
        InitEnemyData();
    }
```
만약 나중에 생성된 인스턴스를 Destroy를 하려면 제일 먼저 생성된 인스턴스를 파괴하지 않게 처리해야지 참조 에러가 생기지 않아요.